server/runtime/src/main/java/org/infinispan/server/context/ServerInitialContextFactoryBuilder.java

```diff
@@ -0,0 +1,35 @@
+package org.infinispan.server.context;
+
+import java.io.Closeable;
+import java.util.Hashtable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import javax.naming.spi.InitialContextFactory;
+import javax.naming.spi.InitialContextFactoryBuilder;
+
+import org.infinispan.commons.util.Util;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 11.0
+ **/
+public class ServerInitialContextFactoryBuilder implements InitialContextFactoryBuilder, Closeable {
+
+   private final ConcurrentMap<String, Object> NAMED_OBJECTS;
+
+   public ServerInitialContextFactoryBuilder() {
+      NAMED_OBJECTS = new ConcurrentHashMap<>();
+   }
+
+   @Override
+   public InitialContextFactory createInitialContextFactory(Hashtable<?, ?> environment) {
+      return new ServerInitialContextFactory(NAMED_OBJECTS);
+   }
+
+   @Override
+   public void close() {
+      // Closes any AutoCloseable objects stored in the context
+      NAMED_OBJECTS.values().stream().filter(v -> v instanceof AutoCloseable).forEach(o -> Util.close((AutoCloseable) o));
```

Comment:
AFAICT `NAMED_OBJECTS` is always empty.

Reply:
No, it will be filled by calls to ServerInitialContext.bind